gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -25,5 +34,602 @@
+
+    private final String selector;
+
+    JobField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
+      return selector;
+    }
+
+    static String selector(JobField... fields) {
+      HashSet<String> fieldStrings = Sets.newHashSetWithExpectedSize(fields.length + 2);
+      fieldStrings.add(JOB_REFERENCE.selector());
+      fieldStrings.add(CONFIGURATION.selector());
+      for (JobField field : fields) {
+        fieldStrings.add(field.selector());
+      }
+      return com.google.common.base.Joiner.on(',').join(fieldStrings);
+    }
+  }
+
+  /**
+   * Class for specifying dataset list options.
+   */
+  class DatasetListOption extends Option {
+
+    private static final long serialVersionUID = 8660294969063340498L;
+
+    private DatasetListOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the maximum number of datasets to be returned.
+     */
+    public static DatasetListOption maxResults(long maxResults) {
+      return new DatasetListOption(BigQueryRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /**
+     * Returns an option to specify the page token from which to start listing datasets.
+     */
+    public static DatasetListOption startPageToken(String pageToken) {
+      return new DatasetListOption(BigQueryRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an options to list all datasets, even hidden ones.
+     */
+    public static DatasetListOption all() {
+      return new DatasetListOption(BigQueryRpc.Option.ALL_DATASETS, true);
+    }
+  }
+
+  /**
+   * Class for specifying dataset get, create and update options.
+   */
+  class DatasetOption extends Option {
+
+    private static final long serialVersionUID = 1674133909259913250L;
+
+    private DatasetOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to specify the dataset's fields to be returned by the RPC call. If this
+     * option is not provided all dataset's fields are returned. {@code DatasetOption.fields}) can
+     * be used to specify only the fields of interest. {@link DatasetInfo#datasetId()} is always
+     * returned, even if not specified.
+     */
+    public static DatasetOption fields(DatasetField... fields) {
+      return new DatasetOption(BigQueryRpc.Option.FIELDS, DatasetField.selector(fields));
+    }
+  }
+
+  /**
+   * Class for specifying dataset delete options.
+   */
+  class DatasetDeleteOption extends Option {
+
+    private static final long serialVersionUID = -7166083569900951337L;
+
+    private DatasetDeleteOption(BigQueryRpc.Option option, Object value) {
+      super(option, value);
+    }
+
+    /**
+     * Returns an option to delete a dataset even if non-empty. If not provided, attempting to
+     * delete a non-empty dataset will result in a {@link BigQueryException} being thrown.
```

Comment:
Ok, it would be nice to include that in the Javadoc (though can wait until you tie BigQueryError to the exception).